### PR TITLE
Fix New User Free Fio Handle Scene

### PR DESCRIPTION
**Pull request by Jon-edge**
image

image

### CHANGELOG

- fixed: FIO Create Handle: Unable to delete first two characters of the handle when using backspace.
- fixed: FIO Create Handle: Double tapping clear in the handle input box removes the domain.

### Dependencies

 none

### Requirements

If you have made **any** visual changes to the GUI. Make sure you have:

- [x] Tested on iOS device
- [ ] Tested on Android device
- [ ] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204385705490826
  - https://app.asana.com/0/0/1204378156120794
  - https://app.asana.com/0/0/1204366217589433

### CODE REVIEW

File: src/components/scenes/Fio/FioCreateHandleScene.tsx [hunk 1]
```diff
@@ -69,12 +69,18 @@ export const FioCreateHandleScene = ({ navigation, route }: Props) => {
   const inputRef = React.useRef<TextInput>(null)
   const mounted = React.useRef<boolean>(true)
 
+  // TODO: Give feedback to the user to indicate why the input was rejected
   const handleChangeFioHandle = useHandler((userInput: string) => {
-    // Clean the userInput:
-    userInput = userInput.replace(domainStr, '').trim()
+    // Dash '-' allowed,
+    // but cannot be the first...
+    if (userInput.indexOf('-') === 0) {
```
Comment (paullinator) [inline src/components/scenes/Fio/FioCreateHandleScene.tsx:76]: Unnecessary change. Could have left the original line 77 and reduce regression potential.

File: src/actions/LoginActions.tsx [hunk 2]
```diff
@@ -73,22 +73,28 @@ export function initializeAccount(navigation: NavigationBase, account: EdgeAccou
...
+
+        // New user FIO handle registration flow (if env is properly configured)
+        const { freeRegApiToken = '', freeRegRefCode = '' } = typeof ENV.FIO_INIT === 'object' ? ENV.FIO_INIT : {}
+        const isCreateHandle =
+          freeRegApiToken !== '' && freeRegRefCode !== '' && (await Airship.show<boolean>(bridge => <FioCreateHandleModal bridge={bridge} />))
+
+        if (isCreateHandle) {
+          // We need to start the wallet creation process before starting the
+          // FIO workflow or wallet creation can fail.
+          createCustomWallets(account, fiatCurrencyCode, selectedEdgetokenIds, dispatch)
```
Comment (paullinator) [inline src/actions/LoginActions.tsx:92]: Definitely want to await this before calling updateWalletsRequest()

File: src/components/themed/OutlinedTextInput.tsx [hunk 3]
```diff
@@ -332,47 +324,48 @@ export const OutlinedTextInput = React.forwardRef<OutlinedTextInputRef, Outlined
...
-            secureTextEntry={hidePassword}
-            // Callbacks:
-            onBlur={handleBlur}
-            onChangeText={onChangeText}
-            onFocus={handleFocus}
-            maxLength={maxLength}
-            onSelectionChange={handleSelectionChange}
-          />
-        )}
+        <View style={[styles.suffixContainer, suffixPadding]}>
```
Comment (paullinator) [inline src/components/themed/OutlinedTextInput.tsx:327]: This affects ALL use of OutlineTextInput and I can't confirm this doesn't cause a regression. I would reduce the regression surface by limiting to textinput, and conditionalize to when `suffix` is being used. Even if it means copy and pasting the entire <TextInput> block, I'd be ok with that and prefer it to affecting all uses.

**Review by paullinator — CHANGES_REQUESTED** [review]

File: src/components/scenes/Fio/FioCreateHandleScene.tsx (hunk 1 above)
Comment (Jon-edge) [reply src/components/scenes/Fio/FioCreateHandleScene.tsx:76]: Line 77 was causing the bug where the user was unable to delete the first two characters via backspace

File: src/actions/LoginActions.tsx (hunk 2 above)
Comment (Jon-edge) [reply src/actions/LoginActions.tsx]: Commit removed

File: src/components/scenes/Fio/FioCreateHandleScene.tsx [hunk 4]
```diff
@@ -71,6 +71,9 @@ export const FioCreateHandleScene = ({ navigation, route }: Props) => {
 
   // TODO: Give feedback to the user to indicate why the input was rejected
   const handleChangeFioHandle = useHandler((userInput: string) => {
+    // Clean spaces:
+    userInput = userInput.replace(domainStr, '').trim()
+
```
Comment (paullinator) [inline src/components/scenes/Fio/FioCreateHandleScene.tsx:76]: why did this get added back. There should be no domain in the string

**Review by paullinator — CHANGES_REQUESTED** [review]

File: src/components/scenes/Fio/FioCreateHandleScene.tsx (hunk 4 above)
Comment (Jon-edge) [reply src/components/scenes/Fio/FioCreateHandleScene.tsx:76]: You're right, intended to ensure spaces were cleaned too but it's already handled.

**Review by paullinator — APPROVED** [review]
